shared/pkg/families/manager.go

```diff
@@ -77,17 +78,43 @@ func New(logger *log.Entry, config *Config) *Manager {
 
 type RunErrors map[types.FamilyType]error
 
-func (m *Manager) Run() (*results.Results, RunErrors) {
-	familyErrors := make(map[types.FamilyType]error)
+type familyResult struct {
+	result interfaces.IsResults
+	err    error
+}
 
+func (m *Manager) Run(ctx context.Context) (*results.Results, RunErrors) {
+	familyErrors := make(RunErrors)
 	familyResults := results.New()
+	result := make(chan familyResult, len(m.families))
 
+outer:
 	for _, family := range m.families {
-		ret, err := family.Run(familyResults)
-		if err != nil {
-			familyErrors[family.GetType()] = fmt.Errorf("failed to run family %v: %w", family.GetType(), err)
-		} else {
-			familyResults.SetResults(ret)
+		go func() {
+			ret, err := family.Run(familyResults)
+			result <- familyResult{
+				ret,
+				err,
+			}
+		}()
+
+	inner:
+		for {
+			select {
+			case <-ctx.Done():
+				log.Warningf("received context cancelled while family %q is still running. Aborting...", family)
+				break outer
+			case r := <-result:
+				log.Debugf("received result from family %q: %v", family, r)
+				if r.err != nil {
+					familyErrors[family.GetType()] = fmt.Errorf("failed to run family %v: %w", family.GetType(), r.err)
+				} else {
+					familyResults.SetResults(r.result)
+				}
+				break inner
+			default:
+				continue
+			}
```

Comment:
we don't want to send the context all the way down to the family/job manager? then each family would return failure

I guess because that logic is in kubeclarity we'd need to make a change there first to add context ?

Reply:
That's correct. I have started the work in KubeClarity, but it will take a decent effort to sort this out there. I wanted to have a short term solution here in order to get the initial version of this functionality done. Otherwise, I'll keep refactoring here and there while the aborting functionality moving nowhere. :)